packages/apollo-gateway/src/datasources/__tests__/RemoteGraphQLDataSource.test.ts

```diff
@@ -195,6 +376,39 @@ describe('didReceiveResponse', () => {
     expect(spyDidReceiveResponse).toHaveBeenCalledTimes(1);
 
   });
+
+  // APQ makes two requests, so make sure only one calls the response hook.
+  it('is only called once when apq is enabled', async () => {
+    class MyDataSource extends RemoteGraphQLDataSource {
+      url = 'https://api.example.com/foo';
+      apq = true;
+
+      didReceiveResponse<MyContext>({
+        response,
+      }: Required<Pick<
+        GraphQLRequestContext<MyContext>,
+          'request' | 'response' | 'context'
+      >>) {
+        return response;
+      }
+    }
+
+    const DataSource = new MyDataSource();
+    const spyDidReceiveResponse = jest.spyOn(DataSource, 'didReceiveResponse');
+
+    fetch.mockJSONResponseOnce({ data: { me: 'james' } });
```

Comment:
Do we need to also simulate the initial miss for this test to be valid?

Via this line which is used in other tests:
`fetch.mockJSONResponseOnce(apqNotFoundResponse);`